code/game/machinery/doors/firedoor.dm

```diff
@@ -88,11 +94,236 @@
+
+		///used so that we can log everyone's stats once when the do_after finishes
+		var/list/user_stat_list = list(list("user" = living_user, "starting temperature" = user_loc_starting_temperature, "starting health" = starting_health, "starting time" = starting_time))
+		people_trying_to_open += user_stat_list
+		RegisterSignal(living_user, COMSIG_PARENT_PREQDELETED, .proc/remove_from_opening_list)
+
+		///we dont want people with only one hand to have the same visible_message as people with two hands
+		var/hand_string = "hands"
+		if(iscarbon(living_user))
+			var/mob/living/carbon/carbon_user = living_user
+			var/obj/item/bodypart/right_arm = carbon_user.get_bodypart(BODY_ZONE_R_ARM)
+			var/obj/item/bodypart/left_arm = carbon_user.get_bodypart(BODY_ZONE_L_ARM)
+
+			if(!right_arm || right_arm.bodypart_disabled)//this is assuming that you have at least one hand since... this proc is called attack_hand
+				hand_string = "hand"
+			if(!left_arm || left_arm.bodypart_disabled)
+				hand_string = "hand"
+
+		var/user_their_pronoun = user.p_their()
+		if(ishuman(user))
+			var/mob/living/carbon/human/human_user = user
+			user_their_pronoun = human_user.p_their()
+
+		//we want to acknowledge that several people are trying to open the same firelock to encourage them to do it
+		//also to signal to people doing it by themselves that its faster with help
+		if(length(people_trying_to_open) == 1)
+			user.visible_message("<span class='notice'>[living_user] tries to open [src] with [user_their_pronoun] [hand_string], struggling greatly to open the heavy door by themselves.</span>", \
+				"<span class='notice'>You try with all your strength to pry open [src] with your [hand_string], barely moving [p_them()]!</span>")
+		else if (length(people_trying_to_open) == 2)
+			user.visible_message("<span class='notice'>[living_user] joins [people_trying_to_open[1]["user"]] in trying to open [src] with [user_their_pronoun] [hand_string].</span>", \
+				"<span class='notice'>You join [people_trying_to_open[1]] in trying to pry open [src] with your [hand_string]!</span>")
+		else
+			user.visible_message("<span class='notice'>[living_user] joins the others in trying to open [src] with [user_their_pronoun] [hand_string]!</span>", \
+				"<span class='notice'>You join the others in trying to pry open [src] with your [hand_string]!</span>")
+
+		//stops the do_after if the firelock is open
+		var/datum/callback/is_closed_callback = CALLBACK(src, .proc/is_closed)
+
+		//this callback adjusts the timer of the do_after_dynamic for each user if the number of people trying to open the firelock changes
+		var/datum/callback/timer_callback = CALLBACK(src, .proc/adjust_do_after_timer)
+
+		START_PROCESSING(SSmachines, src)//burns the arms of everyone trying to open the firelock if its hot enough and they dont have protection
+
+		//TODOKYLER: the progress bar is inaccurate for teaming up, and only the first person will succeed, this needs to be centralized to work
```

Comment:
Look at this TODO.

Reply:
oops, forgot i added that after i finished the others